control_toolbox/src/control_filters/gravity_compensation.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2023, Stogl Robotics Consulting UG (haftungsbeschränkt)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "control_filters/gravity_compensation.hpp"
+
+#include "geometry_msgs/msg/vector3_stamped.hpp"
+#include "geometry_msgs/msg/wrench_stamped.hpp"
+#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
+#include "tf2/LinearMath/Vector3.h"
+
+namespace control_filters
+{
+template <>
+bool GravityCompensation<geometry_msgs::msg::WrenchStamped>::update(
+  const geometry_msgs::msg::WrenchStamped & data_in, geometry_msgs::msg::WrenchStamped & data_out)
+{
+  if (!this->configured_)
+  {
+    RCLCPP_ERROR_SKIPFIRST_THROTTLE((*logger_), *clock_, 2000, "Filter is not configured");
+    return false;
+  }
```

Comment:
```suggestion
```

Reply:
we have this check in the other filters as well:
https://github.com/ros-controls/control_toolbox/blob/3ed348a6736d763b0170d0bb3ee3d8fe0b0e42a3/control_toolbox/include/control_filters/exponential_filter.hpp#L110-L113

But in the other filters we throw an exception, instead of logging. Should we change it?

Reply:
Sure. Let's do as other filters :)

Reply:
done

Reply:
What I can say is that when I was asked to integrate the parameters with the generate_parameter_library, I did try to change the minimum possible while checking the code was doing what it was supposed to (hence fixing also the computation at some point), but the original PR had a return with false so I kept it.  
Having an "OROCOS" background, where there is a state machine, a component could not run if it was not configured properly, so the run/update phase was not supposed to check again if the controller was configured, but the whole idea was to not crash/end the whole control loop due to deployment (load unload) at runtime.  I forgot if the controller manager correctly catches those throw.  If there is a mechanism capturing, then a thrown might be fine.

Reply:
The controller_manager handles this (we removed those checks in the controllers), but this filter could be used stand-alone as well. Then it's the responsibility of the user to take care of that

Reply:
We decided to throw an exception because if someone forgot to configure the plugin and tried to run update directly, it shouldn't continue returning false as it is not functional or recoverable in this state, so we thought throwing an exception is the right idea

Reply:
thanks for the clarification. I am glad to see this work is going to be integrated. Always cool to see something being used.